lib/cache/add-local.js

```diff
@@ -92,13 +93,21 @@ function addLocalDirectory (p, pkgData, shasum, cb) {
     getCacheStat(function (er, cs) {
       mkdir(path.dirname(pj), function (er, made) {
         if (er) return cb(er)
-        var doPrePublish = !pathIsInside(p, npm.tmp)
-        if (doPrePublish) {
-          lifecycle(data, 'prepublish', p, iferr(cb, thenPack))
-        } else {
-          thenPack()
+        var isProduction = npm.config.get('production')
+        var dir = npm.dir
+        var prefix = npm.prefix
+
+        npm.config.set('production', false)
```

Comment:
This stuff, in particular, is extremely scary and probably shouldn't be allowed in production code.